kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/config/VirtualCluster.java

```diff
@@ -56,6 +59,11 @@ public record VirtualCluster(@NonNull @JsonProperty(required = true) String name
     public VirtualCluster {
         Objects.requireNonNull(name);
         Objects.requireNonNull(targetCluster);
+        if (!isDnsLabel(name)) {
```

Comment:
I feel like this is overly restrictive. Can we not have it fail if there clustername to injected into the pattern is not a valid DNS label? I guess thats hard as we don't know at config parsing time we we actually need to inject the name.

Reply:
I'll see what we can do, it wouldn't be at config parsing time but I think the other branch could do some validation

Reply:
the injection does happen somewhere where an exception will prevent the proxy starting, so we could potentially put this requirement there. It's not at config parsing time but the user will find out about it fast.

Though it feels untidy to have an exception like "your cluster name must be in format X to use feature Y". Having the name limitation up front is easier to describe "cluster names must have format: ".

Reply:
But it's only feature Y that requires the format rules. I agree it's not a great experience but it does limit the restriction to the subset of users who it matters to.